src/components/FlowRunTimeline.vue

```diff
@@ -309,11 +309,45 @@
     api.flowRuns.getFlowRuns,
     subFlowRunsSubscriptionFilter,
   )
+
+  const subFlowsSubscriptionFilter = computed<Parameters<typeof api.flows.getFlows> | null>(() => {
+    if (!subFlowRunsSubscription.response || subFlowRunsSubscription.response.length < 1) {
+      return null
+    }
+
+    const subFlowIds = subFlowRunsSubscription.response
+      .map((flowRun) => flowRun.flowId)
+
+    return [
+      {
+        flows: {
+          id: subFlowIds,
+        },
+      },
+    ]
+  })
+
+  const subFlowSubscription = useSubscriptionWithDependencies(
+    api.flows.getFlows,
+    subFlowsSubscriptionFilter,
+  )
+
   const subFlowRunLabels = computed(() => {
     return (subFlowRunsSubscription.response ?? [])
       .reduce((acc, curr) => {
         if (curr.name) {
-          acc.set(curr.id, curr.name)
+          let subFlowRunName = ''
+
+          if (subFlowSubscription.response) {
+            const subFlow = subFlowSubscription.response.find((flow) => flow.id === curr.flowId)
```

Comment:
Do you have an idea of how many subflows might exist? If there are a lot we probably should convert the response to a map keyed by the id so we can do lookups like this without looping.